File: frontend/src/services/openapi/models/CompetitionStatusContentModel.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { CurrentCompetitorContentModel } from './CurrentCompetitorContentModel';
import type { DivisionStatusModel } from './DivisionStatusModel';

/**
 * Current status of the competition.
 */
export type CompetitionStatusContentModel = {
    /**
     * Name of the event. Example: National finals 2022
     */
    eventName: string;
    /**
     * Timestamp indicating when this status was generated.
     * This is always In UTC
     * Format "yyyy-MM-ddTHH:mm:ss.fffZ"
     */
    createdAt: string;
    /**
     * Current status of divisions
     */
    divisions: Array<DivisionStatusModel>;
    currentCompetitor?: CurrentCompetitorContentModel;
}

File: src/Api.Models/CompetitionStatusContentModel.cs
using System.ComponentModel.DataAnnotations;

namespace Api.Models
{
    /// <summary>
    /// Current status of the competition.
    /// </summary>
    public class CompetitionStatusContentModel
    {
        /// <summary>
        /// Name of the event. Example: National finals 2022
        /// </summary>
        [Required]
        public string EventName { get; set; } = null!;

        /// <summary>
        /// Timestamp indicating when this status was generated
        /// </summary>
        [Required]
        public string CreatedAt { get; set; } = null!;

        /// <summary>
        /// Current status of divisions
        /// </summary>
        [Required]
        public DivisionStatusModel[] Divisions { get; set; } = null!;

        /// <summary>
        /// Current competitor. May be null if not set.
        /// </summary>
        public CurrentCompetitorContentModel? CurrentCompetitor { get; set; }
    }
}
